pkg/manifests/manifests.go

```diff
@@ -2878,6 +2878,39 @@ func (f *Factory) ControlPlanePrometheusRule() (*monv1.PrometheusRule, error) {
 		}
 		r.Spec.Groups = groups
 	}
+	if !f.infrastructure.HighlyAvailableInfrastructure() {
```

Comment:
the control plane topology is also interesting and should be `||` in this condition, I think some of the aggregated APIs are scheduled on the control plane nodes

Reply:
Added a new function to check for this.